### ISSUE TITLE
[BUG] <title>

### ISSUE THREAD

**Issue by ibolton336** [konveyor/tackle2-ui#1824]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

latest

### Priority

Critical

### Current Behavior

The function localeNumericCompare() has proven to be a valuable utility for comparing strings numerically in a way that respects locale-specific ordering. 

This issue is to cover standardizing its usage across the application

### How Reproducible

Always (Default)

### Environment

```markdown
- OS:
```

**Comment by sjd78** [comment]
@rszwajko - Did your natural sort ordering PR also take care of this issue?

**Comment by rszwajko** [comment]
> @rszwajko - Did your natural sort ordering PR also take care of this issue?

No. The `selectionOptions` need to be explicitly sorted  - in the same way we do it for tag categories here -> https://github.com/konveyor/tackle2-ui/blob/56e556f23d54b7ecb656e1345e63ddf65695c757/client/src/app/pages/applications/components/application-tags/application-tags.tsx#L151
